packages/react-ui/src/hooks/useSort.ts

```diff
@@ -18,13 +19,15 @@ export default function useSort(
     return `${sort.direction === "desc" ? "-" : ""}${sort.sortBy}`
   }, [sort.sortBy, sort.direction])
 
-  const updateSort = (sortBy: string) => {
-    if (sort.sortBy === undefined || sort.sortBy !== sortBy) {
-      setSort({ sortBy, direction: "asc" })
+  const updateSort = (sortBy: string, alwaysSorted = sort.alwaysSorted) => {
+    if (alwaysSorted && sort.direction === "desc") {
+      setSort({ sortBy, direction: "asc", alwaysSorted })
+    } else if (sort.sortBy === undefined || sort.sortBy !== sortBy) {
+      setSort({ sortBy, direction: "asc", alwaysSorted })
     } else if (sort.direction === "asc") {
-      setSort({ sortBy, direction: "desc" })
+      setSort({ sortBy, direction: "desc", alwaysSorted })
     } else {
-      setSort({ sortBy: undefined, direction: undefined })
+      setSort({ sortBy: undefined, direction: undefined, alwaysSorted })
```

Comment:
similarly here, if `alwaysSorted` is passed in as `true`, should the other arguments be defined?

Reply:
No if `alwaysSorted` is true, it won't take action until the `direction` is being changed from `desc`

Reply:
👍